### ISSUE TITLE
Indexes get applied wrongly when using inheritance

### ISSUE THREAD

**Issue by Ben305** [typegoose/typegoose#194]
## Versions
- NodeJS: 10.18.0
- Typegoose(NPM): 6.3.1
- Typegoose(GIT): commithash
- mongoose: 5.8.11
- mongodb: 0.0.0

## What is the Problem?

I'm using the index decorators on multiple classes, but after the first models have been generated, indexes get applied to other classes which shouldn't get these indexes.

## Code Example

I try to prepare a sample, but let me describe this:

```ts
@index({ deleted: 1 })
class Base {
    @prop()
    deleted: false;
}

class ClassWithNoIndex extends Base {
    @prop()
    anotherProperty: '';
}

@index({ category: 1 })
class ClassWithOtherIndexes extends Base {
    @prop()
    category: 0;
}

class AnotherClassWithoutIndexes extends Base {
    @prop()
    myProp: '';
}
```

After calling `getModelForClass` for all these classes, the model prepared for `AnotherClassWithoutIndexes` also gets indexes from `ClassWithOtherIndexes`.

## Do you know *why* it happenes?

*no*

**Comment by hasezoey** [comment]
just to confirm, it happens even if they are in the same file? (to exclude some import problems - havnt tested yet)

**Comment by Ben305** [comment]
All classes are located in different files. I didn't copy & paste the sample but typed it.

**Comment by hasezoey** [comment]
i interpreted your code, made an working example, made models, made proper types, and annotated what indexes i got - and yes, i noticed that some have indexes they should have (all should have "deleted" since it is in the base model)

```ts
// NodeJS: 12.15.0
// MongoDB: 4.2-bionic (Docker)
import { getModelForClass, prop, index } from "@typegoose/typegoose"; // @typegoose/typegoose@6.3.1
import * as mongoose from "mongoose"; // mongoose@5.8.11

@index({ deleted: 1 })
class Base { // i take this as an "abstract-like" class that dosnt have an model
  @prop({ default: false })
  public deleted!: boolean;
}

class ClassWithNoIndex extends Base { // has "category" and "deleted"
  @prop({ default: "" })
  public anotherProperty!: string;
}

@index({ category: 1 })
class ClassWithOtherIndexes extends Base { // has "category" and "deleted"
  @prop({ default: 0 })
  public category!: number;
}

class AnotherClassWithoutIndexes extends Base { // has "deleted"
  @prop({ default: "" })
  public myProp!: string;
}

const ClassWithNoIndexModel = getModelForClass(ClassWithNoIndex);
const ClassWithOtherIndexesModel = getModelForClass(ClassWithOtherIndexes);
const AnotherClassWithoutIndexesModel = getModelForClass(AnotherClassWithoutIndexes);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verify194", useCreateIndex: true, useUnifiedTopology: true });

  const doc1 = await ClassWithNoIndexModel.create({});
  const doc2 = await ClassWithOtherIndexesModel.create({});
  const doc3 = await AnotherClassWithoutIndexesModel.create({});

  console.log(doc1, doc2, doc3);

  await mongoose.disconnect();
})();
```

**Comment by hasezoey** [comment]
found the issue, will be fixed in 6.3.2
